client/src/components/CoursePane/CourseRenderPane.js

```diff
@@ -142,36 +142,27 @@ class CourseRenderPane extends PureComponent {
             };
 
             try {
-                const response = await queryWebsoc(params);
-
-                if (response.ok) {
-                    const jsonResp = await response.json();
-
-                    const adBannerInfo = await fetch(
-                        `${RANDOM_AD_ENDPOINT}?deptCode=${encodeURIComponent(formData.deptValue)}`
-                    );
-
-                    if (adBannerInfo.ok) {
-                        const jsonAdInfo = await adBannerInfo.json();
-
-                        this.setState({
-                            loading: false,
-                            error: false,
-                            courseData: flattenSOCObject(jsonResp),
-                            bannerName: jsonAdInfo.bannerName,
-                            bannerLink: jsonAdInfo.bannerLink,
-                        });
-                    } else {
-                        this.setState({
-                            loading: false,
-                            error: false,
-                            courseData: flattenSOCObject(jsonResp),
-                        });
-                    }
+                const jsonResp = await queryWebsoc(params);
+
+                const adBannerInfo = await fetch(
+                    `${RANDOM_AD_ENDPOINT}?deptCode=${encodeURIComponent(formData.deptValue)}`
+                );
+
+                if (adBannerInfo.ok) {
+                    const jsonAdInfo = await adBannerInfo.json();
+
+                    this.setState({
+                        loading: false,
+                        error: false,
+                        courseData: flattenSOCObject(jsonResp),
+                        bannerName: jsonAdInfo.bannerName,
+                        bannerLink: jsonAdInfo.bannerLink,
+                    });
                 } else {
                     this.setState({
                         loading: false,
-                        error: true,
+                        error: false,
+                        courseData: flattenSOCObject(jsonResp),
```

Comment:
This will probably be a merge conflict with #239.  
Let's close that one first and then merge the changes into this branch.